site/gatsby-site/src/components/SemanticallyRelatedIncidents.js

```diff
@@ -45,11 +46,18 @@ const SemanticallyRelatedIncidents = ({ incident, editable }) => {
 
   const [error, setError] = useState(null);
 
+  const initialDisplay = useRef(true);
+
+  const { setFieldValue } = editable ? useFormikContext() : { setFieldValue: null };
```

Comment:
This violates one of the rules of hooks. See `do not call hooks inside conditions.`

(It may work in this case, but it is a code smell).

Reply:
I don't see a good way to avoid this. `useFormikContext()` will throw an exception if it's not inside a Formik form. On the submissions list, we display the similar incidents to users without editing permission, so it's not in a form. The only way I can think of to avoid making this conditional is to just make a completely separate component for non-editable lists of related incidents. But that would be non-DRY and I think would smell worse.

Reply:
That's what HOCs are for. You export the component with `setFieldValue=null` and then export it again using `withFormik`.

Reply:
@lmcnulty after improving this one and removing the logs I'll approve.

Reply:
Fixed.